tests/versioning.go

```diff
@@ -1677,16 +1677,16 @@ func (s *VersioningIntegSuite) waitForPropagation(ctx context.Context, tq, newBu
 	nsId := s.getNamespaceID(s.namespace)
 	s.Eventually(func() bool {
 		for pt := range remaining {
-			partName, err := tqname.FromBaseName(tq)
+			taskQueue, err := tqid.FromBaseName("", taskQueue)
 			s.NoError(err)
-			partName = partName.WithPartition(pt.part)
+			partition := taskQueue.NormalPartition(0, pt.part)
 			// Use lower-level GetTaskQueueUserData instead of GetWorkerBuildIdCompatibility
 			// here so that we can target activity queues.
 			res, err := s.testCluster.host.matchingClient.GetTaskQueueUserData(
 				ctx,
 				&matchingservice.GetTaskQueueUserDataRequest{
 					NamespaceId:   nsId,
-					TaskQueue:     partName.FullName(),
+					TaskQueue:     partition.RpcName(),
 					TaskQueueType: pt.tp,
```

Comment:
and then you can use `partition.TaskType()` here? (except please rename that to TaskQueueType)